pandas/tests/base/test_unique.py

```diff
@@ -100,12 +98,11 @@ def test_nunique_null(null_obj, index_or_series_obj):
 
 
 @pytest.mark.single_cpu
-@pytest.mark.xfail(using_string_dtype(), reason="decoding fails")
 def test_unique_bad_unicode(index_or_series):
     # regression test for #34550
     uval = "\ud83d"  # smiley emoji
 
-    obj = index_or_series([uval] * 2)
+    obj = index_or_series([uval] * 2, dtype=object)
```

Comment:
Why is the solution here to add `dtype=object`? Shouldn't this just work naturally with the inferred string type?

Reply:
This doesn't work with a string dtype because this test is about "bad unicode". And an actual string dtype cannot represent invalid unicode (at least when using pyarrow under the hood. I assume that our object-dtype based one will be able to hold it). 

To keep the spirit of the test (ensure our unique implementation can work with bad unicode in object dtype), I made it explicitly used object dtype.

See also the "Invalid unicode input" section in https://github.com/pandas-dev/pandas/issues/59328 (that issue I started yesterday to start record breaking changes / things that are no longer supported with the string dtype)

Reply:
Makes sense - also makes me think how we can leverage a BinaryDtype in the future, though that is a different topic for a different day